javalanglib/src/main/scala/java/lang/System.scala

```diff
@@ -207,11 +235,40 @@ object System {
     }
   }
 
-  //def getProperties(): java.util.Properties
-  //def getProperty(key: String): String
-  //def getProperty(key: String, default: String): String
-  //def clearProperty(key: String): String
-  //def setProperty(key: String, value: String): String
+  def getProperties(): ju.Properties = {
+    // Option(getSecurityManager()).foreach(_.checkPropertiesAccess)
+    systemProperties
+  }
+
+  def setProperties(properties: ju.Properties): Unit = {
+    // Option(getSecurityManager()).foreach(_.checkPropertiesAccess)
+    if (properties == null)
+      throw new scala.NullPointerException
+    systemProperties = properties
+  }
+
+  def getProperty(key: String): String = {
+    // Option(getSecurityManager()).foreach(_.checkPropertiesAccess)
+    systemProperties.getProperty(key)
+  }
+
+  def getProperty(key: String, default: String): String = {
+    // Option(getSecurityManager()).foreach(_.checkPropertiesAccess)
+    systemProperties.getProperty(key, default)
+  }
+
+  def clearProperty(key: String): String = {
+    // Option(getSecurityManager()).foreach(_.checkPermission(ju.PropertyPermission(key, "write")))
+    systemProperties.remove(key).asInstanceOf[String]
+  }
+
+  def setProperty(key: String, value: String): String = {
+    // Option(getSecurityManager()).foreach(_.checkPermission(ju.PropertyPermission(key, "write")))
+    systemProperties.setProperty(key, value).asInstanceOf[String]
+  }
+
+  // If implemented, commented code in property related methods must be uncommented
+  //  def getSecurityManager(): SecurityManager
```

Comment:
This is misleading. There are zillions of places in the already implemented parts of the JDK that assume no security manager. This comment suggests that only the above methods will have to be updated, but that's not true.

I recommend removing all comments related to security managers.